mex/extractors/open_data/extract.py

```diff
@@ -68,3 +74,37 @@ def extract_open_data_persons_from_open_data_parent_resources(
             for person in (resource.metadata.creators + resource.metadata.contributors)
         }
     )
+
+
+def extract_tableschema(version_id: int) -> dict[str, list[OpenDataTableSchema]]:
+    """Extract the metadata zip tableschemas.
+
+    Args:
+        version_id: id of record version as integer
+
+    Returns:
+        tableschema by name of tableschema json
+    """
+    connector = OpenDataConnector.get()
+
+    zip_file = connector.get_schema_zipfile(version_id)
+
+    schema_collection: dict[str, list[OpenDataTableSchema]] = {}
+    with ZipFile(BytesIO(zip_file.content)) as zf:
+        schema_file_paths = [
+            n
+            for n in zf.namelist()
+            if (
+                n.lower().startswith("metadaten/schemas/tableschema_")
+                or n.lower().startswith("schemas/tableschema_")
+            )
+            and n.lower().endswith(".json")
+        ]
+        for file_path in schema_file_paths:
+            with zf.open(file_path) as f:
+                data = OpenDataTableSchemaJson.model_validate(
+                    json.load(TextIOWrapper(f, encoding="utf-8"))
+                )
+            schema_collection[file_path.split("schemas/", 1)[1]] = data.fields
```

Comment:
```suggestion
            schema_collection[file_path.split("schemas/", 1)[1]] = data
```
I think it would be a little cleaner to return the highest level pydantic model `OpenDataTableSchemaJson`. Otherwise drop that model entirely and validate `OpenDataTableSchema` in a loop.

Reply:
[34d45d7](https://github.com/robert-koch-institut/mex-extractors/pull/103/commits/34d45d7c63bc598d6e6039dc76b8fdaee4a146f1)